com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -1230,6 +1228,7 @@ private void HandleRawTransportPoll(NetworkEvent networkEvent, ulong clientId, A
 #if DEVELOPMENT_BUILD || UNITY_EDITOR
                     s_TransportDisconnect.Begin();
 #endif
+                    OnClientDisconnectCallback?.Invoke(clientId);
```

Comment:
The only reservation I have about this is that this allows users to still try to send messages.
Really, it might help to have two callbacks:
OnClientDisconnecting
<NetworkManager Shutdown>
OnClientDisconnected  
 =or= 
We maybe need a OnNetworkManagerShutdown callback?

Reply:
The above is just my thoughts... no need to make additional adjustments.
This actually will fix some other issues regarding parenting and clients disconnecting.